### ISSUE TITLE
Improve Developer Testing

### ISSUE THREAD

**Issue by omad** [opendatacube/odc-tools#521]
In **ODC Tools** we have a several automated tests written for some of the tools and libraries. However, having recently tried to do some work on the code here, I've run across a few hiccups.

**Problems**
- Most tests required an available PostgreSQL database to be setup.
- Interdependencies between tests, many tests have been written such that they depend on previous tests being run. This makes it impossible to run individual tests during development.

The tests have been setup with [pytest-depends](https://pypi.org/project/pytest-depends/), but it's being used in a manner other than advised.

From the third paragraph documentation of **pytest-depends**:
> This isn't a dependency in the sense of test A sets up stuff for test B, but more in the sense of if test A failed there's no reason to bother with test B either.

Two different options for development setups have been documented in [README.md#Local Development](https://github.com/opendatacube/odc-tools#local-development), but this does nothing to solve the test interdependency problem to be able to run individual tests, and is awkward at best in automating a PostgreSQL setup.

**Comment by omad** [comment]
**Another Issue**

Tests written for the AWS capable tools depend on S3 resources hosted by third parties, and some of those have recently become unavailable. Some of the STAC documents store in `s3://sentinel-cogs/` are no longer publicly readable.

I tried replacing the reads of real S3 documents with [moto](https://github.com/spulec/moto), but failed due to the use of [aiobotocore](https://github.com/aio-libs/aiobotocore), see: #522 .

An alternate possible solution for this could be using a mock server after implementing a solution to #470 . But using [moto](https://github.com/spulec/moto) would have been simpler.

**Comment by omad** [comment]
Done
